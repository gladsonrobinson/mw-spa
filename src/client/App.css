.main-container {
  text-align: center;
}

.main-container .header {
  background-color: #282c34;
  min-height: 10px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.main-container .header a {
  color: #ffffff;
  margin: 0 20px;
}

.container {
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}
@media (min-width: 768px) {
  .container {
    max-width: 738px;
  }
}
@media (min-width: 992px) {
  .container {
    max-width: 940px;
  }
}
@media (min-width: 1200px) {
  .container {
    max-width: 1170px;
  }
}

.container.mw-map {
  display: flex;
  flex-direction: row;
  margin-top: 10px;
}
