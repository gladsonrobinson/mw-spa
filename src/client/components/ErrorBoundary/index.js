import React from "react";

export default class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasComponentError: false };
  }

  componentDidCatch(error, info) {
    this.setState({ hasComponentError: true });
  }

  render() {
    if (this.state.hasComponentError) {
      return <h4 className="component--error">Some thing went wrong</h4>
    }
    return this.props.children;
  }
}
