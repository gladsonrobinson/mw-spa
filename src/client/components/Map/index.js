import React from "react";
import { compose, withProps } from "recompose"
import { withScriptjs, withGoogleMap, GoogleMap, Marker } from "react-google-maps"

const Map = compose(
  withProps({
    googleMapURL: "https://maps.googleapis.com/maps/api/js?key=AIzaSyBY448uB6sUyz3ETiIYwciXQdpAOoWGeGQ&v=3.exp&libraries=geometry,drawing,places",
    loadingElement: <div style={{ height: `100%` }} />,
    containerElement: <div style={{ height: `400px`, flex: 1 }} />,
    mapElement: <div style={{ height: `100%` }} />,
  }),
  withScriptjs,
  withGoogleMap
)((props) =>
  <GoogleMap
    defaultZoom={5}
    defaultCenter={{ lat: 51.1657, lng: 10.4515 }}
  >
    {
      props.mapMarkers.length && 
      props.mapMarkers.map(item => 
        <Marker position={{ lat: item.latLng.lat, lng: item.latLng.lng }} key={item._id} /> )
    }
  </GoogleMap>
);

export default Map;
