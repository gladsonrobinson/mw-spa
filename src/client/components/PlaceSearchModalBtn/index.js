import React from "react";
import ReactModal from "react-modal";
import LocationSearchInput from "../LocationSearchInput";
import "./PlaceSearchModalBtn.css";

export default class PlaceSearchModalBtn extends React.Component {
  render() {
    return (
      <div className="add-map">
        <button
          className="add-map__button"
          onClick={this.props.handleOpenModal}
        >
          ADD MAP
        </button>

        <ReactModal isOpen={this.props.showModal} contentLabel="Place Search">
          <div>
            <button onClick={this.props.handleCloseModal}>Close Modal</button>

            <LocationSearchInput onSelect={this.props.onPlaceSelect} />
          </div>
        </ReactModal>
      </div>
    );
  }
}
