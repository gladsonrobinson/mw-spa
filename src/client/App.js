import React, { Component } from "react";
import "./App.css";
import { Provider } from "react-redux";
import store from "./redux/store";
import AppCtnr from "./containers/App";

export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <AppCtnr />
      </Provider>
    );
  }
}
