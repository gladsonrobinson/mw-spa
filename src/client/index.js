import React from "react";
import ReactDOM from "react-dom";
import "./index.css";
import App from "./App";
console.log("map key", process.env.REACT_APP_MAP_KEY)

ReactDOM.render(<App />, document.getElementById("root"));
