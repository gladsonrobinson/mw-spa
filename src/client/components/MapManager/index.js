import React from "react";
import PlaceSearchModalBtn from "../PlaceSearchModalBtn";
import "./map-manager.css";

export default class MapManager extends React.PureComponent {
  onDelectClick = id => {
    this.props.removeMapMarker(id);
  };
  render() {
    return (
      <div className="map-manager">
        <div>
          <PlaceSearchModalBtn
            showModal={this.props.showModal}
            handleOpenModal={this.props.showPlaceSearchModal}
            handleCloseModal={this.props.hidePlaceSearchModal}
            onPlaceSelect={this.props.onPlaceSelect}
          />
        </div>
        <div className="map-manager-markers">
          {this.props.mapMarkers.length > 0 &&
            this.props.mapMarkers.map(item => {
              return (
                <div className="map-manager-markers__item" key={item._id}>
                  <h4 className="map-manager-markers__item__address">
                    {item.address}
                  </h4>
                  <div className="map-manager-markers__item__latlng">
                    Latitude: {item.latLng.lat}
                  </div>
                  <div className="map-manager-markers__item__latlng">
                    Longitude: {item.latLng.lng}
                  </div>
                  <button
                    className="map-manager-markers__item__delete"
                    onClick={this.onDelectClick.bind(null, item._id)}
                  >
                    Delete
                  </button>
                </div>
              );
            })}
        </div>
      </div>
    );
  }
}
