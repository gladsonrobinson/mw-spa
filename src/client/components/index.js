import Wrapper from "./Wrapper";
import SearchBox from "./SearchBox";
import Map from "./Map";
import MapManager from "./MapManager";

export { Wrapper, SearchBox, Map, MapManager };
