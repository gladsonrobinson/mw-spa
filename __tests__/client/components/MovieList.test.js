import React from "react";
import { MovieList } from "../../../src/client/components";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";

describe("Movie List Component", () => {
  it("Should render the component MovieList", () => {
    const wrapper = shallow(<MovieList />);
    expect(wrapper.exists()).toBe(true);
  });

  it("Should not render any items if list is empty", () => {
    const movies = [];
    const wrapper = shallow(<MovieList movies={movies} />);
    expect(wrapper.find(".movie-list")).toBeDefined();
    expect(wrapper.find(".movie-list__item")).toHaveLength(0);
  });

  it("Should render correctly when list is empty", () => {
    const wrapper = renderer.create(<MovieList movies={[]} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should not break without movies props", () => {
    const wrapper = shallow(<MovieList />);
    expect(wrapper.find(".movie-list")).toBeDefined();
    expect(wrapper.find(".movie-list__item")).toHaveLength(0);
  });

  it("Should render the complete list of items", () => {
    const movies = [
      {
        Title: "CSI: Crime Scene Investigation",
        Year: "2000–2015",
        imdbID: "tt0247082",
        Type: "series",
        Poster:
          "https://images-na.ssl-images-amazon.com/images/M/MV5BMTkyODgwMDMzNV5BMl5BanBnXkFtZTgwMTExOTMyMjE@._V1_SX300.jpg"
      },
      {
        Title: "American Crime Story",
        Year: "2016–",
        imdbID: "tt2788432",
        Type: "series",
        Poster:
          "https://m.media-amazon.com/images/M/MV5BNzc2MzJmM2ItMjgzYy00MjgxLTljYjctZjJhYzM1ODFhMzU0XkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SX300.jpg"
      }
    ];
    const wrapper = shallow(<MovieList movies={movies} />);
    expect(wrapper.find(".movie-list")).toBeDefined();
    expect(wrapper.find(".movie-list__item")).toHaveLength(movies.length);
  });

  it("Should render correctly when list have items", () => {
    const movies = [
      {
        Title: "CSI: Crime Scene Investigation",
        Year: "2000–2015",
        imdbID: "tt0247082",
        Type: "series",
        Poster:
          "https://images-na.ssl-images-amazon.com/images/M/MV5BMTkyODgwMDMzNV5BMl5BanBnXkFtZTgwMTExOTMyMjE@._V1_SX300.jpg"
      },
      {
        Title: "American Crime Story",
        Year: "2016–",
        imdbID: "tt2788432",
        Type: "series",
        Poster:
          "https://m.media-amazon.com/images/M/MV5BNzc2MzJmM2ItMjgzYy00MjgxLTljYjctZjJhYzM1ODFhMzU0XkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SX300.jpg"
      }
    ];
    const wrapper = renderer.create(<MovieList movies={movies} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render results when the movie changes", () => {
    const wrapper = mount(<MovieList movies={[]} />);
    wrapper.setProps({
      movies: [
        {
          Title: "CSI: Crime Scene Investigation",
          Year: "2000–2015",
          imdbID: "tt0247082",
          Type: "series",
          Poster:
            "https://images-na.ssl-images-amazon.com/images/M/MV5BMTkyODgwMDMzNV5BMl5BanBnXkFtZTgwMTExOTMyMjE@._V1_SX300.jpg"
        }
      ]
    });
    expect(wrapper.find("img").prop("src")).toEqual(
      "https://images-na.ssl-images-amazon.com/images/M/MV5BMTkyODgwMDMzNV5BMl5BanBnXkFtZTgwMTExOTMyMjE@._V1_SX300.jpg"
    );
  });
});
