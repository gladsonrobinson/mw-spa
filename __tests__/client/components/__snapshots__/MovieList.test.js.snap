// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie List Component Should render correctly when list have items 1`] = `
<ul
  className="movie-list"
>
  <li
    className="movie-list__item"
  >
    <img
      alt="Poster"
      className="movie-list__item__img"
      src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTkyODgwMDMzNV5BMl5BanBnXkFtZTgwMTExOTMyMjE@._V1_SX300.jpg"
    />
    <div
      className="movie-list__item__tittle"
    >
      CSI: Crime Scene Investigation
    </div>
  </li>
  <li
    className="movie-list__item"
  >
    <img
      alt="Poster"
      className="movie-list__item__img"
      src="https://m.media-amazon.com/images/M/MV5BNzc2MzJmM2ItMjgzYy00MjgxLTljYjctZjJhYzM1ODFhMzU0XkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SX300.jpg"
    />
    <div
      className="movie-list__item__tittle"
    >
      American Crime Story
    </div>
  </li>
</ul>
`;

exports[`Movie List Component Should render correctly when list is empty 1`] = `
<ul
  className="movie-list"
/>
`;
